pep-9999.rst

```diff
@@ -0,0 +1,501 @@
+backwards compatibility implications for users who have pinned versions of the
+current ``toml`` PyPI package. For more details, see `<Alternative names for
+module_>`_.
+
+
+Security Implications
+=====================
+
+Errors in the implementation could cause potential security issues.
+The parser's output is limited to simple data types; inability to load
+arbitrary classes avoids security issues common in more "powerful" formats like
+pickle and YAML. Also, the implementation will be in pure Python, which reduces
+security issues endemic to C, such as buffer overflows.
+
+
+How to Teach This
+=================
+
+The API of ``tomllib`` mimics that of other well-established file format
+libraries, such as ``json`` and ``pickle``. The lack of a ``dump`` function will
+be explained in the documentation, with a link to relevant third-party libraries
+(``tomlkit``, ``tomli-w``, ``pytomlpp``).
+
+
+Reference Implementation
+========================
+
+The proposed implementation can be found at https://github.com/hukkin/tomli
+
+
+Rejected Ideas
+==============
+
+Basing on another TOML implementation
+-------------------------------------
+
+Potential alternatives include:
+
+* ``tomlkit``.
+  ``tomlkit`` is well established, actively maintained and supports TOML v1. An
+  important difference is that ``tomlkit`` supports style roundtripping. As a
+  result, it has a more complex API and implementation (about 5x as much code as
+  ``tomli``). The author does not believe that ``tomlkit`` is a good choice for
+  the standard library.
+
+* ``toml``.
+  ``toml`` is a widely used library. However, it is not actively maintained,
+  does not support TOML v1 and has several known bugs. Its API is more complex
+  than that of ``tomli``. It has some very limited and mostly unused ability to
+  preserve style through an undocumented decoder API. It has the ability to
+  customise output style through a complicated encoder API. For more details on
+  API differences to this PEP, refer to `Appendix A`_.
+
+* ``pytomlpp``.
+  ``pytomlpp`` is a Python wrapper for the C++ project ``toml++``. Pure Python
+  libraries are easier to maintain than extension modules.
+
+* ``rtoml``.
+  ``rtoml`` is a Python wrapper for the Rust project ``toml-rs`` and hence has
+  similar shortcomings to ``pytomlpp``.
+  In addition, it does not support TOML v1.
+
+* Writing from scratch.
+  It's unclear what we would get from this: ``tomli`` meets our needs and the
+  author is willing to help with its inclusion in the standard library.
+
+Including an API for writing TOML
+---------------------------------
+
+There are several reasons to not include an API for writing TOML:
+
+The ability to write TOML is not needed for the use cases that motivate this
+PEP: for core Python packaging use cases or for tools that need to read
+configuration.
+
+Use cases that involve editing TOML (as opposed to writing brand new TOML) are
```

Comment:
Do you refer for cases like `poetry add` here? For example, `poetry add requests` will write `requests = "^x.y"` to your `pyproject.toml` file.

Reply:
Yes, that's a good example.